lib/exchanges.js

```diff
@@ -0,0 +1,94 @@
+var base      = require('taskcluster-base');
+var assert    = require('assert');
+var _         = require('lodash');
+
+// Common schema prefix
+var SCHEMA_PREFIX_CONST = 'http://schemas.taskcluster.net/github/v1/';
+
+/** Build common routing key construct for `exchanges.declare` */
+var commonRoutingKey = function(options) {
+    options = options || {};
+     let routingKey = [
+        {
+          name:             'routingKeyKind',
+          summary:          "Identifier for the routing-key kind. This is " +
+                            "always `'primary'` for the formalized routing key.",
+          constant:         'primary',
+          required:         true
+        }, {
+          name:             'organization',
+          summary:          "The GitHub `organization` which had an event.",
+          maxSize:          100,
+          required:         true
+        }, {
+          name:             'repository',
+          summary:          "The GitHub `repository` which had an event.",
```

Comment:
Hmm, note the good (and only good thing) about replacing dot for `%` is that you keep the maximum length of 100 characters. Otherwise a github repository consisting of too many dots cannot be represented, because all routing key entries combined must be less than 255 chars + separators...